directory/directory_dest.go

```diff
@@ -116,6 +70,63 @@ func newImageDestination(sys *types.SystemContext, ref dirReference) (private.Im
 	return d, nil
 }
 
+func initDestDir(dir string) error {
+	const versionContents = "Directory Transport Version: 1.1\n"
+	versionFile := filepath.Join(dir, "version")
+
+	// If dir exists, checks if it is empty.
+	// If not empty, check whether the contents match that of a container
+	// image directory and overwrite the contents.
+	// If the contents don't match, throw ErrNotContainerImageDir.
+	fd, err := os.Open(dir)
+	switch {
+	case err == nil: // Directory exists.
+		contents, err := fd.Readdirnames(-1)
+		_ = fd.Close()
+		if err != nil { // Unexpected error.
+			return fmt.Errorf("%q: %w", dir, err)
+		}
+		if len(contents) == 0 {
+			break
+		}
+		// Check if contents of version file is what we expect it to be.
+		ver, err := os.ReadFile(versionFile)
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				return ErrNotContainerImageDir
+			}
+			return fmt.Errorf("%q: %w", versionFile, err)
+		}
+		if !bytes.Equal(ver, []byte(versionContents)) {
+			return ErrNotContainerImageDir
+		}
+		// Indeed an image directory. Reuse by removing all the old contents
+		// (including the versionFile, which will be recreated below).
```

Comment:
It seems worth it to preserve the rationale “so that only one image is in the directory at a time”.

(OTOH https://github.com/containers/skopeo/issues/1237 exists, but that’s clearly not _this_ PR.)